param/reactive.py

```diff
@@ -209,17 +248,32 @@ def updating(self):
         self._watch(lambda e: wrapper.param.update(object=False), precedence=999)
         return wrapper.param.object.rx()
 
-    def when(self, *dependencies):
+    def when(self, *dependencies, initial=Undefined):
         """
         Returns a reactive expression that emits the contents of this
-        expression only when the condition changes.
+        expression only when the dependencies change. If initial value
+        is provided and the dependencies are all param.Event types the
+        expression will not be evaluated until the first event is
+        triggered.
 
         Arguments
         ---------
         dependencies: param.Parameter | rx
           A dependency that will trigger an update in the output.
+        initial: object
+          Object that will stand in for the actual value until the
+          first time a param.Event in the dependencies is triggered.
         """
-        return bind(lambda *_: self.value, *dependencies).rx()
+        deps = [p for d in dependencies for p in resolve_ref(d)]
+        is_event = all(isinstance(dep, Event) for dep in deps)
+        def eval(*_, evaluated=[]):
+            if is_event and initial is not Undefined and not evaluated:
```

Comment:
Is it correctly understood that this if-statement should only be run once?

Reply:
Yep.